pdata/README.md

```diff
@@ -0,0 +1,214 @@
+# Pipeline data (pdata)
+
+Pipeline data (pdata) implements data structures that represent telemetry data in-memory. All data received 
+is converted into this format, travels through the pipeline in this format, and is converted from this format by 
+exporters when sending.
+
+Current implementation primarily uses OTLP protobuf structs as the underlying data structures for many of the 
+declared structs. We keep a pointer to OTLP protobuf in the "orig" member field. This allows efficient translation 
+to/from OTLP wire protocol. The underlying data structure is kept private so that we are free to make changes to it 
+in the future.
+
+API provided by pdata ensures that no underlying data can be used in more than one place. Each pdata instance cannot 
+contain a reference to an object that is already used somewhere else.
+
+## API naming convention
+
+### Protobuf message representation in pdata
+
+Packages of pdata are not following names of the protobuf packages. pdata has a package per telemetry type starting 
+with `p`, e.g. `ptace`, and `pcommon` package with API for other protobuf definitions from `common` and `resource` 
+protobuf packages.
+
+Pipeline data structs SHOULD be based on the names of the underlying OTLP protobuf messages. Data types for 
+protobuf messages defined as part of another message SHOULD include the owner's name as a prefix. Exceptions to these 
```

Comment:
Can we add one example to make clear what this means?

Reply:
Added